ORStools/gui/ORStoolsDialog.py

```diff
@@ -605,40 +615,208 @@ def _linetool_annotate_point(
 
         annotation.setFrameSizeMm(QSizeF(8, 5))
         annotation.setFrameOffsetFromReferencePointMm(QPointF(1.3, 1.3))
-        annotation.setMapPosition(point)
         annotation.setMapPositionCrs(crs)
+        annotation.setMapPosition(point)
 
-        return QgsMapCanvasAnnotationItem(annotation, self.annotation_canvas).annotation()
+        return QgsMapCanvasAnnotationItem(annotation, self.canvas).annotation()
 
     def _clear_annotations(self) -> None:
         """Clears annotations"""
-        for annotation_item in self.annotation_canvas.annotationItems():
+        for annotation_item in self.canvas.annotationItems():
             annotation = annotation_item.annotation()
             if annotation in self.project.annotationManager().annotations():
                 self.project.annotationManager().removeAnnotation(annotation)
         self.annotations = []
+        if self.rubber_band:
+            self.rubber_band.reset()
 
     def _on_linetool_init(self) -> None:
         """Hides GUI dialog, inits line maptool and add items to line list box."""
-        # Remove blue lines (rubber band)
-        if self.line_tool:
-            self.line_tool.canvas.scene().removeItem(self.line_tool.rubberBand)
-
         self.hide()
-        self.routing_fromline_list.clear()
-        # Remove all annotations which were added (if any)
         self._clear_annotations()
-
-        self.line_tool = maptools.LineTool(self._iface.mapCanvas())
-        self._iface.mapCanvas().setMapTool(self.line_tool)
-        self.line_tool.pointDrawn.connect(
-            lambda point, idx: self._on_linetool_map_click(point, idx)
+        self.routing_fromline_list.clear()
+        self.line_tool = maptools.LineTool(self.canvas)
+        self.canvas.setMapTool(self.line_tool)
+        self.line_tool.pointPressed.connect(lambda point: self._on_movetool_map_press(point))
+        self.line_tool.pointReleased.connect(
+            lambda point, idx: self._on_movetool_map_release(point, idx)
         )
-        self.line_tool.doubleClicked.connect(self._on_linetool_map_doubleclick)
+        self.line_tool.doubleClicked.connect(self._on_line_tool_map_doubleclick)
+        self.line_tool.mouseMoved.connect(lambda pos: self.change_cursor_on_hover(pos))
+
+    def change_cursor_on_hover(self, pos):
```

Comment:
It feels like these functions (change_cursor_on_hover, check_annotation_hover, _on_movetool_map_press, _on_movetool_map_release, create_rubber_band, create_vertex) should all be part of the line_tool?

Instead, the rubber band has been moved to the main UI - what is the reason for this?